devito/finite_differences/finite_difference.py

```diff
@@ -16,7 +16,7 @@
 @check_input
 @check_symbolic
 def first_derivative(expr, dim, fd_order=None, side=centered, matvec=direct,
-                     symbolic=False):
+                     symbolic=False, x0={}):
```

Comment:
this needs to be added to the docstring. Also, default value should be None I recommend. See this: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
reminder (@rhodrin )